pkg/templates/untrustedimages/template.go

```diff
@@ -0,0 +1,55 @@
+package untrustedimages
+
+import (
+	"fmt"
+	"regexp"
+
+	"golang.stackrox.io/kube-linter/pkg/check"
+	"golang.stackrox.io/kube-linter/pkg/config"
+	"golang.stackrox.io/kube-linter/pkg/diagnostic"
+	"golang.stackrox.io/kube-linter/pkg/objectkinds"
+	"golang.stackrox.io/kube-linter/pkg/templates"
+	"golang.stackrox.io/kube-linter/pkg/templates/untrustedimages/internal/params"
+	"golang.stackrox.io/kube-linter/pkg/templates/util"
+	v1 "k8s.io/api/core/v1"
+)
+
+const (
+	templateKey = "untrusted-images"
+)
+
+func init() {
+	templates.Register(check.Template{
+		HumanName:   "Untrusted images",
+		Key:         templateKey,
+		Description: "Flag applications running containers with untrusted container images",
+		SupportedObjectKinds: config.ObjectKindsDesc{
+			ObjectKinds: []string{objectkinds.DeploymentLike},
+		},
+		Parameters:             params.ParamDescs,
+		ParseAndValidateParams: params.ParseAndValidate,
+		Instantiate: params.WrapInstantiateFunc(func(p params.Params) (check.Func, error) {
+			return util.PerContainerCheck(func(container *v1.Container) []diagnostic.Diagnostic {
+
+				// checking whether the container image is a trusted or not
+
+				allowedResult, blockedResult := false, true
+
+				for _, allowed := range p.AllowedList {
+					allowedRe := regexp.MustCompile(allowed)
```

Comment:
Don't use `MustCompile` here, as that will make `kube-linter` crash at runtime if someone uses a bad regex. Instead, we should compile the regexes at the top of this function, and return an instantiation error. For example, see the pattern [here](https://github.com/stackrox/kube-linter/commit/72b90689e78007030b8aa18513db28a5ec455761#diff-f5177d5dde82cfc5a418885d5ece6774652f07cf0e7ff174d41d06eee1ae169dR41).

Reply:
Thanks for pointing. I'll move it to top & add proper error handling.